react/index.js

```diff
@@ -51,21 +52,94 @@ class MiniCart extends Component {
   state = {
     openContent: false,
     updatingOrderForm: false,
+    offline: false,
+  }
+
+  updateStatus = () => {
+    if (navigator) {
+      this.setState({ offline: !pathOr(true, ['onLine'], navigator) })
+    }
+  }
+
+  saveDataIntoLocalStorage = () => {
+    if (localStorage) {
+      const clientItems = this.getModifiedItemsOnly()
+      localStorage.setItem('minicart', JSON.stringify(clientItems))
+      const clientOrderForm = pathOr(
+        path(['data', 'orderForm'], this.props),
+        ['linkState', 'orderForm'],
+        this.props
+      )
+      localStorage.setItem('orderForm', JSON.stringify(clientOrderForm))
+    }
+  }
+
+  getDataFromLocalStorage = () => {
+    try {
+      if (localStorage) {
+        const minicart = JSON.parse(localStorage.getItem('minicart'))
+        const orderForm = JSON.parse(localStorage.getItem('orderForm'))
+        return { minicart, orderForm }
+      }
+      return {}
+    } catch (err) {
+      return {}
+    }
+  }
+
+  async componentDidMount() {
+    if (window) {
+      window.addEventListener('online', this.updateStatus)
+      window.addEventListener('offline', this.updateStatus)
+    }
+    this.updateStatus()
+
+    if (localStorage) {
+      const { minicart, orderForm } = this.getDataFromLocalStorage()
+      if (orderForm && !path(['data', 'orderForm'], this.props)) {
+        await this.props.updateOrderForm(orderForm)
+        localStorage.removeItem('orderForm')
+      }
+      if (minicart && minicart.length) {
+        await this.props.addToLinkStateCart(minicart)
+        localStorage.removeItem('minicart')
+      }
+    }
+  }
+
+  componentWillUnmount() {
+    if (window) {
+      window.removeEventListener('online', this.updateStatus)
+      window.removeEventListener('offline', this.updateStatus)
+    }
   }
 
   async componentDidUpdate(prevProps) {
-    await this.handleItemsUpdate()
-    this.handleOrderFormUpdate(prevProps)
+    if (!this.state.offline) {
+      await this.handleItemsUpdate()
+      this.handleOrderFormUpdate(prevProps)
+      if (localStorage) {
+        localStorage.clear()
```

Comment:
I think it's better to clear just the items you're setting. This could cause an unexpected side-effect in other apps that  use localStorage.

Reply:
That's right! I didn't realize it (actually I had this in my mind [here](https://github.com/vtex-apps/minicart/pull/135/files#diff-8587e609aaa7b1140e400929368c45dfR101) but forgot to use it there)